docker/data-science/notebooks/ch07/2.semantic-search.ipynb

```diff
@@ -710,73 +689,61 @@
     "            case _:\n",
     "                additional_query = \"+{!edismax v=\\\"\" + escape_quotes_in_query(item[\"surface_form\"]) + \"\\\"}\"\n",
     "        if additional_query:\n",
-    "            query_tree[i] = {\"type\": \"solr\", \"query\": additional_query}                    \n",
-    "    return query_tree\n",
-    "\n",
-    "def process_semantic_functions(query_tree):\n",
-    "    position = 0\n",
-    "    while position < len(query_tree):\n",
-    "        item = query_tree[position]        \n",
-    "        # process commands. For now, going left to right and then sorting by priority when ambiguous commands occur; \n",
-    "        # consider other weighting options later.\n",
-    "        if (item['type'] == \"semantic_function\"):\n",
-    "            commandIsResolved = False\n",
-    "    \n",
-    "            command = item['semantic_function']\n",
-    "\n",
-    "            if (command):\n",
-    "                query = {\"query_tree\": query_tree} #pass by-ref\n",
-    "                commandIsResolved = eval(item['semantic_function']); #Careful... there is code in the docs that is being eval'd. \n",
-    "                #MUST ENSURE THESE DOCS ARE SECURE, OTHERWISE THIS WILL INTRODUCE A POTENTIAL SECURITY THREAT (CODE INJECTION)\n",
-    "            \n",
-    "            #else:\n",
-    "                #Alert (\"Error: \" + query.query_tree.canonical_form + \" has no command function.\");\n",
-    "            \n",
-    "            if (False == commandIsResolved):\n",
-    "                #Bad command. Just remove for now... could alternatively keep it and run as a keyword\n",
-    "                query_tree.pop(position) #.splice(position,1)  \n",
-    "\n",
-    "        position += 1\n",
-    "\n",
-    "    return query_tree "
+    "            query_tree[i] = {\"type\": \"transformed\",\n",
+    "                             \"syntax\": \"solr\",\n",
+    "                             \"query\": additional_query}                 \n",
+    "    return query_tree"
    ]
   },
   {
    "cell_type": "code",
-   "execution_count": 42,
+   "execution_count": 30,
    "metadata": {
     "tags": []
    },
-   "outputs": [],
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "[{'type': 'engine', 'query': '+{!func v=\"mul(if(stars_i,stars_i,0),20)\"}'}, {'type': 'skg_enriched', 'enrichments': {'category': 'Korean', 'term_vector': 'kimchi^0.9193 korean^0.7069 banchan^0.6593'}}, {'type': 'engine', 'query': '+{!geofilt d=50 sfield=\"location_p\" pt=\"35.22709,-80.84313\"}'}]\n",
```

Comment:
similar to last coment. I think we settled on "{type:transformed, syntax:solr}", but the code seems to have a mix of "engine" and "transformed" for type.